Project02Eclipse/src/com/hemebiotech/analytics/WriteSymptomsResultInFile.java

```diff
@@ -0,0 +1,63 @@
+package com.hemebiotech.analytics;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class contains a method which browse a list of symptoms,
+ * counts the occurrences of each symptom and write the result to a file
+ * @author jonathan.G
+ * @since 1.0
+ */
+public class WriteSymptomsResultInFile implements ISymptomsWriter{
+	
+	private List<String> arraySymptoms;
+	
+	/**
+	 * Constructor
+	 * @param arraySymptoms A list of symptoms
+	 */
+	public WriteSymptomsResultInFile(List<String> arraySymptoms) {
+		this.arraySymptoms = arraySymptoms;
+	}
+	
+	/**
+	 * Extract the differents symptoms and count their occurrences
+	 * Generates a file containing the list of symptoms and the number of times they appear in the file
+	 * @throws IOException 
+	 */
+	public void WriteSymptoms() throws IOException {
+		FileWriter writer = new FileWriter("result.out");
```

Comment:
Avoid declaration inside a function. You should have this one as a property on class level, and inject it in the constructor. (like the `arraySymptoms` property you did).

Reply:
It's done